TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/GenerateRandomNestedJsonObject.java

```diff
@@ -0,0 +1,90 @@
+package org.opensearch.migrations.replay;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.opensearch.migrations.PruferTreeGenerator;
+
+import java.util.AbstractMap;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class GenerateRandomNestedJsonObject {
+
+    public static final String KEY_PREFIX = "keyPrefix-";
+    public static final String VALUE_PREFIX = "VALUE_";
+    ObjectMapper objectMapper;
+
+
+    public GenerateRandomNestedJsonObject() {
+        objectMapper = new ObjectMapper();
+    }
+
+    public String getRandomTreeFormattedAsString(boolean pretty, int seed, int numNodes, int numArrays)
+            throws JsonProcessingException
+    {
+        Random random = new Random(seed);
+        var jsonObj = makeRandomJsonObject(random, numNodes, numArrays);
+        if (pretty) {
+            return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonObj);
+        } else {
+            return objectMapper.writeValueAsString(jsonObj);
+        }
+    }
+
+    @SneakyThrows
+    public static Object makeRandomJsonObject(Random random, int numNodes, int numArrays) {
+        assert numArrays < numNodes;
+        PruferTreeGenerator ptg = new PruferTreeGenerator<String>();
+        var edges = IntStream.range(0, numNodes-3).map(x->random.nextInt(numNodes-1)+1).sorted().toArray();
+        var tree = ptg.makeTree(vn -> Integer.toString(vn), edges);
```

Comment:
what does "vn" stand for?

Reply:
```
    public interface NodeValueGenerator<T> {
        T makeValue(int vertexNumber);
    }
```

Reply:
The prufer sequence implicitly defines these vertexNumbers - so they are indeed unique identifiers, but they might not be very useful for all applications, so that's what the value generator is for.

Reply:
okay, gotcha. makes sense